src/components/BuiltinPlayer.vue

```diff
@@ -72,29 +73,24 @@ let stateInterval: any | undefined;
 const updatePlayerState = function () {
   const player_id = props.playerId;
   if (!audioRef.value || !player_id) return;
+  if (webPlayer.timedOutDueToThrotteling()) {
+    // The player timed out due to the browser freezing the timeout!
+    webPlayer.setTabMode(WebPlayerMode.CONTROLS_ONLY, true);
+    return;
+  }
+  let success;
   if (webPlayer.audioSource === WebPlayerMode.BUILTIN) {
     if (playing.value) {
-      api
-        .updateBuiltinPlayerState(player_id, {
-          powered: true,
-          playing: !audioRef.value.paused,
-          paused: audioRef.value.paused && !audioRef.value.ended,
-          muted: audioRef.value.muted,
-          volume: Math.round(audioRef.value.volume * 100),
-          position: audioRef.value.currentTime,
-        })
-        .catch((e) => {
-          if (e.error_code === 10) {
-            // Player not found exception from backend
-            // this can happen for example if we missed a timeout event from the server
-            // while the browser tab was suspended
-            // temp solution is to simply re-register the player but we might want to
-            // consider a more robust solution in the future
-            api.registerBuiltinPlayer("This Player", player_id);
-          }
-        });
+      success = api.updateBuiltinPlayerState(player_id, {
```

Comment:
```suggestion
      success = await api.updateBuiltinPlayerState(player_id, {
```

this returns a Promise

Reply:
Thanks! Maybe this caused those edge cases I was taking about?

Reply:
yeah, could be